Package/RedPy/redpy.py

```diff
@@ -26,146 +27,121 @@ def download(self, subreddit, number=5, sort_option=None,
-        print(self.url)
+        json_data = self._generateJSON(subreddit, sort_option)
 
-##        session = _requests.Session()
-####        session.headers.update({'User-Agent': self.user})
-##        session.headers.update({'user-agent': 'lmao rofl@matapita.com'})
+        self._DownloadFiles(self._getImages(json_data, number))
 
-        user = {'user-agent':self.user}
-        res = _requests.get(self.url, headers=user)
-
-        if res.status_code != 200:
-            print("Could not download")
-            print(res.status_code)
-            return
-
-        self._DownloadFiles(res.json(), number)
-
-
-    def _DownloadFiles(self, jsonfile, number_of_files):
-        image_links = self._getImages(jsonfile, number_of_files)
+    def _DownloadFiles(self, image_links):
 
         if not self.createFolder():
             print("Error creating folder")
             return
 
-        index = 0 #used to name the files
+        index = 0  # used to name the files
         for image_link in image_links:
             image_link = image_link.replace('amp;', '')
             f = _requests.get(image_link)
-
-            if f.status_code==200:
-                media_file = open(f'{_os.getcwd()}/red_media/{index}.jpg', 'wb')
+            if f.status_code == 200:
+                media_file = open(_os.path.join(
+                    _os.getcwd(), "red_media", f"{index}.jpg"), 'wb')
 
                 for chunk in f.iter_content(100000):
                     media_file.write(chunk)
                 media_file.close()
                 print("Downloaded")
-                index+=1
+                index += 1
         print("Download complete")
-        global flag
-        flag=1
 
-    def download_videos(self, subreddit, number=5, sort_option=None):
-        """Downloads images from subreddit.
+    def _getImages(self, jsonfile, number_of_files):
+
+        images = []  # contains links of images
+        for index in range(number_of_files):
+            try:
+                images.append(jsonfile['data']['children'][index]['data']
+                              ['preview']['images'][0]['source']['url'])
+            except Exception as e:
+                print("Exception: ", e)
+        return images
+
+    def download_videos(self, subreddit, number=5, sort_option=''):
+        """Downloads Videos from subreddit.
             subreddit="Name of subreddit"
-            number=Number of images to be downloaded
+            number=Number of videos to be downloaded
             sort_option=new/hot/top
         """
-        subreddit = subreddit.strip('/')
-        if sort_option == None:
-            sort_option = ''
+        subreddit = subreddit.split('/')[-1] if "/" in subreddit else subreddit
 
-        self.url = 'https://www.reddit.com/r/' + subreddit + '/'+  sort_option  +  '.json'
-        print(self.url)
+        json_data = self._generateJSON(subreddit, sort_option)
```

Comment:
Why are we calling the `_generateJSON` function in both the download media functions? Why not call it once in the download function? Is there any reason you chose this?

Reply:
I think you are doing this because the function is public.

Maybe you can have a default parameter in the function and check its value. If it's empty, then you can call the `generateJSON` function in the separate functions

Reply:
Added `json_data` to member variable so as to call once.
The `_getImages()` and `_getVideos()` still take `json_data` as parameter so as others can parse their own `json_data` if required.
And its used in RedIns as well